.gitlab/includes/common_spack_pipeline.yml

```
@@ -26,3 +26,69 @@ base_spack_image:
   artifacts:
     reports:
       dotenv: base.env
+
+.compiler_image_template: &compiler_image_template
```

Is this needed?

```suggestion
.compiler_image_template:
```
? As I understand it the `&compiler_image_template` makes this into an anchor which is an alternative way to reuse parts of the configuration but I think you're not using that?